### PR TITLE
Bug, Image AltText Duplicate

### PR DESCRIPTION
**Pull request by xichan-ms**
[AB#276643](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/276643)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6411)

### CODE REVIEW

File: docs/specs/validations/metadataContentValidations.yml [hunk 1]
```diff
@@ -2070,24 +2072,21 @@ inputs:
...
-  d.json:
   pig1.jpg:
   pig2.jpg:
+  pig3.jpg:
+  pig4.jpg:
   .errors.log: |
     {"message_severity":"suggestion","log_item_type":"user","code":"image-alt-text-duplicated","message":"Alt text 'alt text' is duplicated. Within an article, alt text must be unique.","file":"a.md","line":2,"end_line":2,"column":1,"end_column":1}
-    {"message_severity":"suggestion","log_item_type":"user","code":"image-alt-text-duplicated","message":"Alt text 'alt text' is duplicated. Within an article, alt text must be unique.","file":"b.md","line":2,"end_line":2,"column":1,"end_column":1}
-    
+    {"message_severity":"suggestion","log_item_type":"user","code":"image-alt-text-duplicated","message":"Alt text 'alt text' is duplicated. Within an article, alt text must be unique. Duplicated altText is in included files: 'a.md(4,1) -> b.md(2,1)'.","file":"b.md","line":2,"end_line":2,"column":1,"end_column":1}
```
Comment (OsmondJiang) [inline docs/specs/validations/metadataContentValidations.yml:2085]: how about report the parent info at main error message just like what heading validation does

Comment (xichan-ms) [reply docs/specs/validations/metadataContentValidations.yml:2085]: OK. Got it.

File: src/docfx/lib/markdown/MarkdownEngine.cs [hunk 2]
```diff
@@ -321,12 +321,12 @@ private string GetImageLink(string path, MarkdownObject origin, string? altText)
                 altText = ToPlainText(origin);
             }
 
-            return GetImageLink(new SourceInfo<string>(path, origin.GetSourceInfo()), altText);
+            return GetImageLink(new SourceInfo<string>(path, origin.GetSourceInfo()), origin, altText);
         }
 
-        private string GetImageLink(SourceInfo<string> href, string? altText)
+        private string GetImageLink(SourceInfo<string> href, MarkdownObject? origin, string? altText)
```
Comment (yufeih) [inline src/docfx/lib/markdown/MarkdownEngine.cs:327]: Should be able to remove all the `?` touched in this PR.

```suggestion
        private string GetImageLink(SourceInfo<string> href, MarkdownObject origin, string? altText)
```

Comment (xichan-ms) [reply src/docfx/lib/markdown/MarkdownEngine.cs]: OK. Got it. Thanks a lot.

**Comment by xichan-ms** [comment]
Diff Explanations:
(1) Some error messages add quotation marks for alt text. Expected.
(2) Some new error messages are duo to the changing of classification file, from currentFile to rootFile. Expected.

**Pull request merged**